docs: Add labels and annotations to created namespaces

philnichol opened the pull request:
Thanks in advance for your review.

This adds the expected labels and annotations to the namespaces that automatically come with a helm installation.

Closes https://github.com/linkerd/linkerd2/issues/7298

I agree to the DCO for all the commits in this PR.

alpeb commented on linkerd.io/content/2.10/tasks/automatically-rotating-webhook-tls-credentials.md:
```diff
@@ -34,12 +34,22 @@ namespace linkerd uses:
 # viz (ignore if not using the viz extension)
 kubectl create namespace linkerd-viz
+kubectl label namespace linkerd-viz linkerd.io/extension=viz
+kubectl annotate namespace linkerd-viz \
+  config.linkerd.io/proxy-await=enabled \
+  linkerd.io/inject=enabled
```
These annotations are not required

alpeb reviewed (commented):
Thanks for the PR @philnichol , and sorry for the late review!

I think we shouldn't push these changes to the 2.9 and 2.10 docs given they might have had different requirements. OTOH would you mind applying them as well to the 2.12 docs that have been merged into this branch?

philnichol commented:
Hey @alpeb no problem at all and thanks for your review.
I've removed the unneeded annotations, reverted the 2.9 and 2.10 docs, and added this to 2.12.
Any issues please let me know.

alpeb reviewed (approved):
Looks good, thanks @philnichol !
